components/encoder/single/single_encoder.go

```diff
@@ -205,7 +206,15 @@ func (e *Encoder) Start(ctx context.Context) {
 				// the motor. This may result in ticks being lost or applied in the wrong direction.
 				dir := e.m.DirectionMoving()
 				if dir == 1 || dir == -1 {
-					atomic.AddInt64(&e.position, dir)
+					tick, err := e.I.Value(ctx, nil)
+					if err != nil {
+						e.logger.Error(err)
+						return
+					}
+					if tick != lastTick {
+						atomic.AddInt64(&e.position, dir)
+					}
+					lastTick = tick
```

Comment:
beautiful. test it out on a motor with a quadrature encoder (on my desk) and the viam-rover to make sure this still works in both cases.

Reply:
do we need a mutex to protect any of this? just noticing.

Reply:
hmm not sure. it didn't seem to have any issues but maybe when reading the tick value on line 209?